### PR TITLE
support tetrahedra in generateStructuredMesh

### PR DESCRIPTION
**Pull request by norihiro-w**
This PR adds support of tetrahedral element type in generateStructuredMesh tool

### CODE REVIEW

File: MeshLib/MeshGenerators/MeshGenerator.h [hunk 1]
```diff
@@ -421,6 +421,62 @@ Mesh* generateRegularPrismMesh(const unsigned n_x_cells,
...
+ * \param div_x Subdivision operator in x direction
+ * \param div_y Subdivision operator in y direction
+ * \param div_z Subdivision operator in z direction
+ * \param origin Optional mesh's origin (the bottom lower left corner) with MathLib::ORIGIN default.
+ */
+Mesh* generateRegularTetMesh(const BaseLib::ISubdivision &div_x,
+                             const BaseLib::ISubdivision &div_y,
+                             const BaseLib::ISubdivision &div_z,
+                             MathLib::Point3d const& origin = MathLib::ORIGIN,
+                             std::string const& mesh_name = "mesh");
```
Comment (endJunction) [inline MeshLib/MeshGenerators/MeshGenerator.h]: mesh_name doxygen parameter doc missing...

**Comment by norihiro-w** [comment]
The suggested code is not generating a correct mesh. Faces of adjacent tetrahedras don't match. I need some more time to fix the issue

**Comment by endJunction** [comment]
After the coordinates are corrected, :fast_forward: from my side...

**Comment by chleh** [comment]
I believe you that you did the splitting right. Code is OK.
Maybe there could be a small comment like "This algorithm generates a structured hex-mesh and splits each hex into six tets."
:+1:

**Comment by norihiro-w** [comment]
Jenkins error is nothing to do with this PR. The error is in NumLib.ComponentNormMultiComponent2

**Comment by norihiro-w** [comment]
travis error is nothing do with this PR. the error is in NumLib.ComponentNormSingleComponent.

**Comment by chleh** [comment]
Then please rebase and merge.

**Comment by TomFischer** [comment]
Could you please also update the documentation on the website: https://docs.opengeosys.org/docs/tools/meshing/structured-mesh-generation?

**Comment by norihiro-w** [comment]
@TomFischer how can I modify the doc?

**Comment by endJunction** [comment]
Never mind the docs. page for now, something is wrong with the logins...

**Comment by endJunction** [comment]
Jenkins, test this please!

**Comment by endJunction** [comment]
Jenkins, test this please!

**Pull request merged**